### PR TITLE
Log to StringIO by default

### LINKED ISSUE #46: higlass-python.log being created everywhere.
Hello~

May I ask for some help regarding a minor issue? higlass-python has been working well (and better and better) for me. However, after I started using higlass-python, each time I create a jupyter notebook in whichever directory, a file named "higlass-python.log" will show up in the same directory. This file is, not surprisingly, empty. I have shut down all the kernels through `Kernel -> Shut Down All Kernels` in jupyter. But still, when I create a new notebook, higlass-python.log will show up. 

This is not a huge issue but just a bit annoying and confusing. May I get some assistance?

Thank you so much!

### PR DESCRIPTION
**Pull request by nvictus**
## Description

What was changed in this pull request?

* Implemented `available-chrom-sizes` endpoint
* `Server`'s logging now defaults to an in-memory stream, stored in `server.log`. Providing a `log_file` will create a rotating file handler instead.
* Since the log's name is inherited from the Flask app name, the Server constructor takes a name parameter, which defaults to `higlass` + a short hash. Getting the logger for a given server will then be `logger = logger.getLogger(server.name)`.

Why is it necessary?

Fixes #46 

## Checklist

- [x] Unit tests added or updated
- [ ] Documentation added or updated
- [x] Updated CHANGELOG.md
- [x] Ran `black` on the root directory

### CODE REVIEW

File: higlass/server.py [hunk 1]
```diff
@@ -24,17 +27,20 @@
...
 log.setLevel(logging.ERROR)
 log.disabled = True
+
 # The following line is also needed to turn off all debug logs
 os.environ["WERKZEUG_RUN_MAIN"] = "true"
 
 
-def create_app(tilesets, name, log_file, log_level, file_ids, fuse=None):
-    app = Flask(__name__)
+def create_app(name, tilesets, fuse=None):
```
Comment (flekschas) [inline higlass/server.py]: Is that API change public? I.e., does a user need to care about it? If not please ignore my comment.

File: higlass/server.py [hunk 2]
```diff
@@ -302,83 +316,101 @@ class Server:
...
     def __init__(
         self,
         tilesets,
-        port=None,
         host="localhost",
-        tmp_dir="/tmp/hgflask",
-        no_fuse: bool = False,
+        port=None,
+        name=None,
+        fuse=True,
```
Comment (flekschas) [inline higlass/server.py]: I am never sure myself but is this common practice? I typically assume if no argument was passed its value is falsy. That's why I called it `no_fuse`. I agree the double negation is also not optimal.

File: higlass/server.py (hunk 1 above)
Comment (nvictus) [reply higlass/server.py]: So far, nobody uses the app factory, except maybe me. Going forward, it is pretty well encapsulated in `Server` anyway.

File: higlass/server.py (hunk 2 above)
Comment (nvictus) [reply higlass/server.py]: I don't think it's uncommon to have flags with true as default. I'm actually anticipating that this becomes False by default in the future, or perhaps changing to a third value like `"infer"` to decide whether to use fuse by scanning the tilesets provided.

File: higlass/server.py (hunk 1 above)
Comment (flekschas) [reply higlass/server.py]: Cool thanks. Was just wondering if this has a big impact or not.

File: higlass/server.py (hunk 2 above)
Comment (flekschas) [reply higlass/server.py]: Cool thanks for clarifying.
> I'm actually anticipating that this becomes False by default in the future
Looking forward to that :)

**Comment by nvictus** [comment]
Done!

File: higlass/server.py [hunk 3]
```diff
@@ -297,88 +311,103 @@ def get_filepath(self, url):
...
+    Parameters
+    ----------
+    tilesets : list
+        A list of tilesets to serve (see higlass.tilesets)
+    host : str, optional
+        The host this server is running on. Usually just localhost.
+    port : int, optional
+        The port that this server will run on.
+    name : str, optional
+        A name for the Flask app being served
```
Comment (pkerpedjiev) [inline higlass/server.py]: Might be useful to have a note as to why one would add this.

**Comment by nvictus** [comment]
> would it be possible to do server.log.info("some info")?

Nope. `server.log` is a file-like buffer backed by a string. The actual logger is controlled by the logging module. I documented how to fetch it.

> Also might be helpful to mention that server in the docs is one of the values returned by higlass.display.

I hope this will eventually not be the case :P I was thinking it should be packaged as an attribute of the viewer object. The viewconf already is.

**Pull request merged**